### PR TITLE
[release-1.7] update go-control-plane

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

Update API definitions for EnvoyFilter support.

### CODE REVIEW

**Comment by kyessenov** [comment]
/retest
@howardjohn sure, will do.

**Comment by kyessenov** [comment]
Holding back names to older version of go-control-plane to avoid breaking changes.

File: pkg/config/validation/validation.go [hunk 1]
```diff
@@ -25,7 +25,6 @@ import (
 	"strings"
 	"time"
 
-	xdsUtil "github.com/envoyproxy/go-control-plane/pkg/wellknown"
```
Comment (mandarjog) [inline pkg/config/validation/validation.go:28]: did these names change back?

**Comment by mandarjog** [comment]
@ramaraochavali can you help review this? some function is broken without the PR, but some tests break *with* the PR.

File: pkg/config/validation/validation.go (hunk 1 above)
Comment (kyessenov) [reply pkg/config/validation/validation.go:28]: Yes, all those names changed. I made all of them literal at the previous SHA of go-control-plane. There is no need to import a changing package just to save on typing a constant.

Comment (mandarjog) [reply pkg/config/validation/validation.go:28]: What happens on the next upgrade?, I suppose as long as don't use the wellknown package we are ok.

Comment (kyessenov) [reply pkg/config/validation/validation.go:28]: We can do whatever want. At some point, istio needs to stop using the deprecated names and APIs.

**Comment by kyessenov** [comment]
@howardjohn All these names are deprecated in Envoy. I don't see the value of creating a package or constants that are going to be removed soon. A simple text search for "envoy." makes it very easy to locate these strings.

**Comment by howardjohn** [comment]
envoy depreciating the names doesn't concern me at all, my concern is that our code relies on the same name being used throughout the codebase (extremely easy to break if we are expecting developers to write the same string in 30 different places) and it makes writing code in pilot a pain - any time I want to name a filter now I'll need to spend a chunk of time figuring out what it should be named, I'd it needs to match any other name, if the name has backwards compatibility issues, etc. I should just be able to type well-known... and let the IDE tell me what to name it

**Comment by kyessenov** [comment]
@howardjohn You shouldn't use names for any new code and you can rely on IDE type completion to fill up protobuf types for typed_config.

**Comment by howardjohn** [comment]
How can I not use a name? We create HCM (for example) in many different
places. Do you suggest each developer picks their favorite name in each
usage instead of a consistent name?

**Comment by ramaraochavali** [comment]
I thought we are doing this only for 1.7. Why do we need to change master - It supports both the names? Is it causing some issue? It seems I am missing some thing here

**Comment by kyessenov** [comment]
Yeah I don't think this should be merged to main. This is to unblock envoy update with as little changes as possible.

**Comment by kyessenov** [comment]
John, I think we need to make a choice for the semantic filter names. I don't think we have to use envoy names at all. Any name would work.

**Comment by ramaraochavali** [comment]
> Any name would work.

Any name does not work in the longer run . As noted here and below https://github.com/istio/istio/pull/25966#issuecomment-666911646 - Envoy might remove the names. If we want our own names, we need to revert that deprecation in Envoy

**Comment by mandarjog** [comment]
I agree we should use semantic names.
I had added references to semantic names in the EnvoyFilter api, those were recently reverted to well known names. 

going forward we should not use hcm  name at all, but we need to support the configuration that is out there.

@ramaraochavali any name should work in most cases.

**Comment by ramaraochavali** [comment]
> going forward we should not use hcm name at all, but we need to support the configuration that is out there.

https://github.com/istio/istio/pull/25966 does the second part but uses wellknown names. If do not want to use wellknown names
- Revert Envoy deprecation of old so that it does not complain about name (that it should be the new canonical name)
- Have our own wellknown with the semantic names we want for standard envoy filters that control plane adds so that every developer does not come up with a new name. If we do this - what is the problem in using Envoy's canonical name for standard filters because users now will have to know about two names while referring subfilters?

**Comment by kyessenov** [comment]
Either way this or #25966. My opinion is that the PR #25966 is more complicated than desired for a SHA bump. But I'm not a maintainer in the area.
I doubt we can un-deprecate things in Envoy. This deprecation is almost a year old now, so the ship has sailed. I would advise not to use these names if possible in the future releases, since they can lead to a crash. For new code, using the updated `wellknown` package is fine with me if you insist on sticking with Envoy implementation names.

**Comment by ramaraochavali** [comment]
My preference is to backport #25966 because it has been there in master for some time now. @howardjohn what do you think?

**Comment by ramaraochavali** [comment]
>  I would advise not to use these names if possible in the future releases, since they can lead to a crash

Do you mean old names?

**Comment by howardjohn** [comment]
> My preference is to backport #25966 because it has been there in master for some time now. @howardjohn what do you think?

+1

**Comment by kyessenov** [comment]
@ramaraochavali yeah, I mean the old names. I think it's risky to change the names in xDS mid-release, but if you think it's fine, backporting sounds fine.

**Comment by ramaraochavali** [comment]
@kyessenov Thanks. I agree on old names. regarding the name changes IMO, it is not that risky (but may be I am biased because I made that change :-) ) .

**Comment by JimmyCYJ** [comment]
What is the status of this PR? Is it still going to 1.7 branch?

**Comment by kyessenov** [comment]
Replaced by https://github.com/istio/istio/pull/26768

**Pull request closed without merging**